lmfit/models.py

```diff
@@ -73,6 +73,7 @@ def guess_from_peak(model, y, x, negative, ampscale=1.0, sigscale=1.0):
 
 def update_param_vals(pars, prefix, **kwargs):
     """Update parameter values with keyword arguments."""
+    pars.update_constraints()
```

Comment:
looking at this a second time, it would probably be better to do `pars.update_constraints()`  after update the parameter values from the keyword arguments.

Reply:
you mean moving it down just before the ```return pars``` in this function? I tried to have only one place of adding this instead of in every model... and this seemed a good option.

Reply:
Yes, right now the constraints get updated, then the variable parameters get new values so the constraints values will be slightly out-of-date.